float64 x
float64 y
---
float64 nx
float64 ny

